src/txkube/test/test_authentication.py

```diff
@@ -68,11 +76,48 @@
 -----END CERTIFICATE-----
 """
 
+# Let hostname u"example.invalid" map to an
+# IPv4 address in the TEST-NET range.
+_hostMap = { u"example.invalid." : "192.0.2.2" }
+
+def createReactor():
+    """
+    Twisted 17.1.0 and higher requires a reactor which implements
+    ``IReactorPluggableNameResolver``.
+    """
+
+    @implementer(IHostResolution)
+    @attr.s
+    class Resolution(object):
+        name = attr.ib()
+
+    class _FakeResolver(object):
+
+        def resolveHostName(self, resolutionReceiver, hostName, *args,  **kwargs):
+            portNumber = kwargs.pop('portNumber')
+            r = Resolution(name=hostName)
+
+            resolutionReceiver.resolutionBegan(r)
+            if hostName in _hostMap:
+                resolutionReceiver.addressResolved(
+                    IPv4Address('TCP', _hostMap[hostName], portNumber))
+            resolutionReceiver.resolutionComplete()
+            return r
```

Comment:
Cool.  This seems to be the right way to get the desired behavior from the new resolver interface.  I'm not quite sure I entirely understand how the interface means to deal with errors - but not calling `addressResolved` seems to be how `_resolver.py` in Twisted deals with errors, so it makes sense to do the same here.